custom_components/adaptive_lighting/switch.py

```diff
@@ -290,6 +294,94 @@ async def handle_set_manual_control(switch: AdaptiveSwitch, service_call: Servic
+    original_sun_light_settings = switch.config_sun_light_settings  # pylint: disable=protected-access
+    if data[CONF_USE_CONFIG_DEFAULTS]:
+        sun_light_settings = original_sun_light_settings
+    else:
+        sun_light_settings = current_sun_light_settings
+
+    astral_location = sun_light_settings.astral_location
+    max_brightness=sun_light_settings.max_brightness if not CONF_MAX_BRIGHTNESS in data else data[CONF_MAX_BRIGHTNESS]
+    max_color_temp=sun_light_settings.max_color_temp if not CONF_MAX_COLOR_TEMP in data else data[CONF_MAX_COLOR_TEMP]
+    min_brightness=sun_light_settings.min_brightness if not CONF_MIN_BRIGHTNESS in data else data[CONF_MIN_BRIGHTNESS]
+    min_color_temp=sun_light_settings.min_color_temp if not CONF_MIN_COLOR_TEMP in data else data[CONF_MIN_COLOR_TEMP]
+    sleep_brightness=sun_light_settings.sleep_brightness if not CONF_SLEEP_BRIGHTNESS in data else data[CONF_SLEEP_BRIGHTNESS]
+    sleep_color_temp=sun_light_settings.sleep_color_temp if not CONF_SLEEP_COLOR_TEMP in data else data[CONF_SLEEP_COLOR_TEMP]
+    sleep_rgb_color=sun_light_settings.sleep_rgb_color if not CONF_SLEEP_RGB_COLOR in data else data[CONF_SLEEP_RGB_COLOR]
+    sleep_rgb_or_color_temp=sun_light_settings.sleep_rgb_or_color_temp if not CONF_SLEEP_RGB_OR_COLOR_TEMP in data else data[CONF_SLEEP_RGB_OR_COLOR_TEMP]
+    sunrise_offset=sun_light_settings.sunrise_offset if not CONF_SUNRISE_OFFSET in data else data[CONF_SUNRISE_OFFSET]
+    if use_actual_sunrise_time:
+        if CONF_SUNRISE_TIME in data:
+            _LOGGER.warn(
+                "ignoring 'sunrise_time' variable due to use_actual_sunrise_time being on. Set to false if you want to specify your own sunrise time. Data '%s'",
+                data,
+            )
+        sunrise_time = None
+    else:
+        sunrise_time=sun_light_settings.sunrise_time if not CONF_SUNRISE_TIME in data else data[CONF_SUNRISE_TIME]
+    max_sunrise_time=sun_light_settings.max_sunrise_time if not CONF_MAX_SUNRISE_TIME in data else data[CONF_MAX_SUNRISE_TIME]
+    sunset_offset=sun_light_settings.sunset_offset if not CONF_SUNSET_OFFSET in data else data[CONF_SUNSET_OFFSET]
+    if use_actual_sunset_time:
+        if CONF_SUNSET_TIME in data:
+            _LOGGER.warn(
+                "ignoring 'sunset_time' variable due to use_actual_sunset_time being on. Set to false if you want to specify your own sunset time. Data '%s'",
+                data,
+            )
+        sunset_time = None
+    else:
+        sunset_time=sun_light_settings.sunset_time if not CONF_SUNSET_TIME in data else data[CONF_SUNSET_TIME]
+    min_sunset_time=sun_light_settings.min_sunset_time if not CONF_MIN_SUNSET_TIME in data else data[CONF_MIN_SUNSET_TIME]
+    transition=sun_light_settings.transition if not CONF_TRANSITION in data else data[CONF_TRANSITION]
+
+    object.__setattr__(switch, '_sun_light_settings', SunLightSettings(  # pylint: disable=protected-access
```

Comment:
Why not
```
switch. _sun_light_settings = SunLightSettings(...)
```

Reply:
I was unsure how python handled the old values in memory.

example:
```
a = ["s": 5]
b = deepcopy(a)
a = None
```

what happens to the original `a`?

I was afraid too many calls to change_sunlight_settings would cause memory leaks (was new to python)